aiohttp/connector.py

```diff
@@ -125,7 +125,13 @@ def __init__(self, *, conn_timeout=None, keepalive_timeout=30,
         self._conns = {}
         self._acquired = defaultdict(set)
         self._conn_timeout = conn_timeout
-        self._keepalive_timeout = keepalive_timeout
+        if force_close and keepalive_timeout:
```

Comment:
This naive check is backward incompatible and will break too many existing code.

I recommend the following:
```
marker = object()

class BaseConnector:
    def __init__(self, keepalive_timeout=marker, force_close=marker):
        if keepalive_timeout is not marker and force_close is not marker:
            if keepalive_timeout and force_close:
                raise ValueError(...)
        elif keepalive_timeout is not marker:
            force_close = ...
        elif force_close is not maker:
            keepalive_timeout = None
```

As a side note after writing a snipped I've figured out that we need to manage `keepalive_timeout=0` also, it should imply `force_close=True` as well.

So my suggestion is: restrict `keepalive_timeout` to `float` only, raise a `TypeError` if `None`.
All other corrections are still make sense.